### PR TITLE
JSON formatting support using spaces for indentation, for issue #2185

### PR DESCRIPTION
**Pull request by rowstop**
### What this PR does / why we need it?
JSON formatting support using spaces for indentation, for issue #2185

#### Please indicate you've done the following:

- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Made sure commit message follow the rule of [Conventional Commits specification](https://www.conventionalcommits.org/).
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed.

### CODE REVIEW

**Comment by wenshao** [comment]
抱歉这就久才处理，我对这个修改有担心，担心因为codeSize变大而使得性能下降

File: core/src/main/java/com/alibaba/fastjson2/JSONWriterUTF8.java [hunk 1]
```diff
@@ -205,10 +205,15 @@ protected final void write0(char c) {
     @Override
     public final void writeColon() {
         int off = this.off;
-        if (off == bytes.length) {
+        if (pretty && off + 1 >= bytes.length) {
```
Comment (wenshao) [inline core/src/main/java/com/alibaba/fastjson2/JSONWriterUTF8.java:208]: 这样的实现，会导致writeColon方法变大很多，从而影响性能

**Comment by rowstop** [comment]
> 抱歉这就久才处理，我对这个修改有担心，担心因为codeSize变大而使得性能下降

👌好的，将来有时间我会考虑做这方面的优化

**Comment by rowstop** [comment]
如果有其它方面更好的想法请告知我，我会在优化时考虑到

**Comment by wenshao** [comment]
https://github.com/alibaba/fastjson2/pull/3225
我也提交了一个PR实现同样的功能。不过PrettyFormat时冒号后面没有加空格，因为这个会改变行为，会导致不兼容。

**Pull request closed without merging**